base62.go

```diff
@@ -1,17 +1,17 @@
 package ksuid
 
-import "bytes"
+import "strings"
 
 // lexographic ordering (based on Unicode table) is 0-9A-Za-z
-var base62Characters = []byte("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz")
+var base62Characters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
 
-func base2base(in []byte, inBase int, outBase int) []byte {
-	res := []byte{}
-	bs := in[:]
+func appendBase2Base(dst []byte, src []byte, inBase int, outBase int) []byte {
```

Comment:
Curios: If the return is dst that has had append() ran on it (presumably causing mutation and a new "value"), how would this save allocations? Is this append() slice magic?